src/software/new_geom/util/distance.cpp

```diff
@@ -19,7 +19,11 @@ double distance(const Point &first, const Point &second)
 
 double distance(const Segment &first, const Segment &second)
 {
-    if (intersects(first, second))
+    boost::geometry::model::segment<Point> AB(first.getSegStart(), first.getEnd());
+    boost::geometry::model::segment<Point> CD(second.getSegStart(),
+                                              second.getEnd());  // similar code
+    bool intersects = boost::geometry::intersects(AB, CD);
+    if (intersects)
```

Comment:
Replaced the usage of existing `intersects` in old geom with its implementation to avoid a circular dependency